mesonbuild/build.py

```diff
@@ -123,13 +125,13 @@ def get_target_macos_dylib_install_name(ld) -> str:
 class InvalidArguments(MesonException):
     pass
 
-class DependencyOverride:
+class DependencyOverride(InterpreterHeldObject):
```

Comment:
I think this is backwards, we should have a `BuildObject` base class defined in the build module, and the interpreter should import that rather tha build being aware of Interpreter. The problem with this approach is that the build module now has to be aware of the interpreter (a layering violation), and one of my big goals in the type work I'm doing in the interpreter is to fix that violation

Reply:
`InterpreterHeldObject` is actually defined in `mesonlib`, so it's technically only the name. See my comment below.